docs/source/groupby-api.rst

```diff
@@ -0,0 +1,19 @@
+API
```

Comment:
I started it there and moved it because I didn't think it made a lot of sense organizationally. How do you want it to look?